packages/react/src/Bento/Bento.tsx

```diff
@@ -0,0 +1,181 @@
+import React, {ReactHTML, ReactElement} from 'react'
+import clsx from 'clsx'
+import {Icon, IconProps} from '@primer/octicons-react'
+import type {BaseProps} from '../component-helpers'
+import {Heading, Text, Link, HeadingProps, TextProps, LinkProps} from '../'
+
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/bento/colors-with-modes.css'
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/bento/base.css'
+import styles from './Bento.module.css'
+
+export type Size = 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge' | 'xxlarge'
+export type ColumnIndex = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
+
+type ResponsiveColumnIndex = Partial<Record<Size, ColumnIndex>> | ColumnIndex
+type ResponsiveRowIndex = Partial<Record<Size, number>> | number
+
+type Flow = 'row' | 'column'
+type ResponsiveFlow = Partial<Record<Size, Flow>> | Flow
+
+type Align = 'start' | 'center' | 'end'
+type ResponsiveAlign = Partial<Record<Size, Align>> | Align
+
+type BentoProps = React.HTMLAttributes<HTMLDivElement> & BaseProps<HTMLDivElement>
+
+const Root = ({className, ...rest}: BentoProps) => {
+  return <section className={clsx(styles.Bento, className)} {...rest}></section>
+}
+
+type BentoItemProps = {
+  columnStart?: ResponsiveColumnIndex
+  columnSpan?: ResponsiveColumnIndex
+  rowStart?: ResponsiveRowIndex
+  rowSpan?: ResponsiveRowIndex
+  flow?: ResponsiveFlow
+  colorMode?: 'light' | 'dark'
+  visualAsBackground?: boolean
+} & React.HTMLAttributes<HTMLDivElement> &
+  BaseProps<HTMLDivElement>
+
+const returnClassBasedOnResponsiveMap = (
+  propName: string,
+  prop?: ResponsiveColumnIndex | ResponsiveRowIndex | ResponsiveFlow | ResponsiveAlign,
+) => {
+  const classesToMerge: string[] = []
+  if (typeof prop === 'number' || typeof prop === 'string') {
+    classesToMerge.push(styles[`Bento__Item--${propName}-${prop}`])
+  } else if (typeof prop === 'object') {
+    for (const [key, value] of Object.entries(prop)) {
+      classesToMerge.push(styles[`Bento__Item--${key}-${propName}-${value}`])
+    }
+  }
+  return classesToMerge
+}
+
+const Item = ({
+  className,
+  columnStart,
+  columnSpan,
+  rowStart,
+  rowSpan,
+  flow,
+  colorMode = 'light',
+  visualAsBackground = false,
+  children,
+  ...rest
+}: BentoItemProps) => {
+  const bentoItemClassArray = [styles.Bento__Item]
+  bentoItemClassArray.push(
+    ...returnClassBasedOnResponsiveMap('column-span', columnSpan),
+    ...returnClassBasedOnResponsiveMap('row-span', rowSpan),
+    ...returnClassBasedOnResponsiveMap('column-start', columnStart),
+    ...returnClassBasedOnResponsiveMap('row-start', rowStart),
+    ...returnClassBasedOnResponsiveMap('flow', flow),
+  )
+
+  return (
+    <div
+      data-color-mode={colorMode}
```

Comment:
Was there a reason for not using ThemeProvider here to set the data-color-mode attribute? Replicating that functionality here makes our color modes more brittle to refactoring, so would be great to use ThemeProvider directly instead.

Reply:
This has been updated in the following SHA: https://github.com/primer/brand/pull/390/commits/015e64c3e342f101c896f8259daf2c49feb49558. Now the functionality should behave as I mentioned above.